src/app/utils/common.ts

```diff
@@ -63,10 +63,17 @@ export const binarySearch = <T>(items: T[], match: (item: T) => -1 | 0 | 1): T |
 export const randomNumberBetween = (min: number, max: number) =>
   Math.floor(Math.random() * (max - min + 1)) + min;
 
-export const scaleYDimension = (x: number, scaledX: number, y: number): number => {
-  const scaleFactor = scaledX / x;
-  return scaleFactor * y;
-};
+export const scaleDimension = (x: number, y: number, maxX: number, minX: number = maxX, maxY: number = Infinity, minY: number = 0): {w: number, h: number} => {
```

Comment:
sure, I just tried to keep all the call-sites relatively unchanged since that made it easier to transition to the new function and debug issues, but now that it's working well I can make it better to read and maintain for the future